code/modules/xenoarchaeology/tools/xenoarchaeology_labeler.dm

```diff
@@ -0,0 +1,261 @@
+	skip_label = TRUE
+	return ..()
+
+/obj/item/xenoarchaeology_labeler/debug/AltClick(mob/user)
+	. = ..()
+	var/choice = tgui_alert(user, "Select Action", "Select Action", list("Toggle Patch", "Change Material"))
+	switch(choice)
+		if("Toggle Patch")
+			patch_traits = !patch_traits
+			to_chat(user, "<span class='notice'>Toggled patch: [patch_traits ? "On" : "Off"].</span>")
+		if("Change Material")
+			var/list/possible_materials = subtypesof(/datum/xenoartifact_material)
+			possible_materials += /datum/xenoartifact_material
+			material = tgui_input_list(user, "Select artifact material.", "Select Material", possible_materials, /datum/xenoartifact_material)
+
+/obj/item/xenoarchaeology_labeler/debug/examine(mob/user)
+	. = ..()
+	. += "<span>Alt+Click to toggle settings.</span>"
+
+/obj/item/xenoarchaeology_labeler/debug/create_label()
+	if(skip_label)
+		skip_label = FALSE
+		return
+	var/obj/item/xenoartifact/no_traits/artifact = new(get_turf(loc))
+	artifact.AddComponent(/datum/component/xenoartifact, material, length(label_traits) ? label_traits : null, TRUE, TRUE, patch_traits)
+
+/*
+	Sticker for labeler, so we can label artifact's with their traits
+*/
+
+/obj/item/sticker/xenoartifact_label
+	icon = 'icons/obj/xenoarchaeology/xenoartifact_sticker.dmi'
+	icon_state = "sticker_star"
+	name = "artifact label"
+	desc = "An adhesive label, for artifacts."
+	do_outline = FALSE
+	roll_unusual = FALSE
+	///List of artifact traits we're labelling
+	var/list/traits = list()
+	///A special examine description built from the traits we have
+	var/examine_override = ""
+	///The original custom price of the item we're going to label
+	var/old_custom_price
+
+/obj/item/sticker/xenoartifact_label/old
+	name = "old artifact label"
+	color = "#bd812e"
+
+/obj/item/sticker/xenoartifact_label/old/build_stuck_appearance()
+	var/mutable_appearance/MA = mutable_appearance(sticker_icon || src.icon, sticker_icon_state || src.icon_state)
+	MA.color = color
+	return MA
+
+
+/obj/item/sticker/xenoartifact_label/Initialize(mapload, list/_traits)
+	ADD_TRAIT(src, TRAIT_ARTIFACT_IGNORE, GENERIC_ITEM_TRAIT)
+	//Setup traits & examine desc
+	traits = _traits.Copy()
+	if(length(traits))
+		examine_override = "Traits:"
+		for(var/datum/xenoartifact_trait/trait_datum as anything in traits)
+			examine_override = "[examine_override]\n	- [initial(trait_datum.label_name)]"
+	//Setup a random appearance
+	icon_state = "sticker_[pick(list("star", "box", "tri", "round"))]"
+	sticker_icon_state = "[icon_state]_small"
+	return ..()
+
+/obj/item/sticker/xenoartifact_label/examine(mob/user)
+	. = ..()
+	. += examine_override
+
+/obj/item/sticker/xenoartifact_label/afterattack(atom/movable/target, mob/user, proximity_flag, click_parameters)
+	//If you somehow make traits start working with mobs, remove this isliving() check
+	if(!isliving(target) && (locate(/obj/item/sticker/xenoartifact_label) in target.contents))
```

Comment:
this still allows mobs to be stickered (i.e. just /mob, /dead, or some abstract things like camera eye)